src/Http/Response.php

```diff
@@ -0,0 +1,238 @@
+     * @param string        $content
+     */
+    public function __construct(
+        array $status = self::STATUS_OK,
+        string $contentType = self::MIME_TEXT_HTML,
+        string $charset = self::CHARSET_UTF8,
+        string $content = ""
+    ) {
+        $this->setStatus($status);
+        $this->contentType = $contentType;
+        $this->charset     = $charset;
+        $this->content     = $content;
+        $this->session     = new Session();
+    }
+
+    /**
+     * @return string
+     */
+    public function getStatusName(): string
+    {
+        return $this->statusName;
+    }
+
+    /**
+     * @param string $statusName
+     */
+    public function setStatusName(string $statusName): void
+    {
+        $this->statusName = $statusName;
+    }
+
+    /**
+     * @return int
+     */
+    public function getStatusCode(): int
+    {
+        return $this->statusCode;
+    }
+
+    /**
+     * @param int $statusCode
+     */
+    public function setStatusCode(int $statusCode): void
+    {
+        $this->statusCode = $statusCode;
+    }
+
+    /**
+     * @param array $status
+     */
+    public function setStatus(array $status): void
+    {
+        $this->statusCode = $status['code'];
+        $this->statusName = $status['name'];
+    }
+
+    /**
+     * @return string
+     */
+    public function getContentType(): string
+    {
+        return $this->contentType;
+    }
+
+    /**
+     * @param string $contentType
+     */
+    public function setContentType(string $contentType): void
+    {
+        $this->contentType = $contentType;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCharset(): string
+    {
+        return $this->charset;
+    }
+
+    /**
+     * @param string $charset
+     */
+    public function setCharset(string $charset): void
+    {
+        $this->charset = $charset;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getContent(): ?string
+    {
+        return $this->content;
+    }
+
+    /**
+     * @param mixed|null $content
+     */
+    public function setContent($content): void
+    {
+        $this->content = $content;
+    }
+
+    /**
+     * @return Bag
+     */
+    public function getData(): Bag
+    {
+        return $this->session->getRequestData();
+    }
+
+    /**
+     * @return Bag
+     */
+    public function getFlashes(): Bag
+    {
+        return $this->session->getFlashes();
+    }
+
+    /**
+     * Content-type is always return
+     * Default: text/html; charset=UTF-8
+     */
+    protected function setContentTypeHeader()
+    {
+        $charset = (substr($this->contentType, 0, 4) === "text") ? "; charset=$this->charset" : null;
+        header("Content-Type: " . $this->contentType . $charset);
+    }
+
+    /**
+     * Status is always return
+     * Default: 200 OK
+     */
+    protected function setStatusHeader(): void
+    {
+        header("Status: $this->statusName", false, $this->statusCode);
```

Comment:
Codacy found an issue: [The use of function header() is discouraged](https://app.codacy.com/gh/pierregaimard/framework3/pullRequest?prid=6299285)